woocommerce/woocommerce-gutenberg-products-block
# Deprecate the woocommerce_shared_settings hook

This is pulled from #3864 and closes #3748

This deprecates `woocommerce_shared_settings` with a notice. Tests have been updated to ignore warnings - we cannot test for the notice yet because WP tests do not support PHPUnit 9.x.

### How to test the changes in this Pull Request:

Add a filter somewhere, for example, main plugin file.

```
add_filter( 'woocommerce_shared_settings', function( $settings ) { return $settings; } );
```

Visit the store. 

Check the debug log:

<img width="835" alt="Screenshot 2021-04-22 at 15 53 30" src="https://user-images.githubusercontent.com/90977/115736146-f798e300-a382-11eb-81d4-5d28b26d2f79.png">

### Dev Note

Blocks surfaced a hook named `woocommerce_shared_settings` which could be used to change, add, or remove settings that are passed from the server to the client. This is now considered deprecated—a warning will be logged to the browser error console if used.

We recommend instead using the Asset Data Registry directly, so for example:

```
use Automattic\WooCommerce\Blocks\Package;
use Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry;

Package::container()->get( AssetDataRegistry::class )->add( $key, $value )
```

Doing this lessons the risk of conflicting or overwriting data, since the filter allows any setting to be modified which could lead to unintentional side effects.

### Changelog

> Replaced usage of the `woocommerce_shared_settings` hook. This will be deprecated.
